### ISSUE TITLE
\lv opens another viewer after using \ll

### ISSUE THREAD

**Issue by Lorenzbi** [lervag/vimtex#227]
Is it possible to disable the viewer on compile with \ll because I want to use \lv and it opens another viewer (mupdf) that supports forward searching (is this how it's supposed to be?) 

edit: this also happens with zathura

**Comment by Lorenzbi** [comment]
This also happens with zathura

**Comment by lervag** [comment]
Yes, this is how it is supposed to be: `\ll` should start compilation and open the viewer. If the viewer is open, then `\lv` should do forward searching, which should be supported by both mupdf and zathura.

Note that mupdf and zathura requires `xdotool` to work properly with vimtex.

**Comment by Lorenzbi** [comment]
That would be perfect but it seems \lv starts a new viewer that "supports" forward search. I must then close the first viewer and can then use \lv without any problems. Maybe \ll starts the viewer differently and xdotool cannot "see" it?

**Comment by lervag** [comment]
I see, this implies there is a bug or that one of the dependencies are not met.

Could you do the following:

1. Start vim/gvim/mvim, open a tex file
2. Do `\lv`
3. Do `\li`
4. Copy the output from `\li` here (I am particularly interested in the viewer information)

It is also interesting to know the vimtex settings you use, and which system you are on, as well as the version of vim (and if you use vim, gvim, or mvim) and the version of xdotools.

**Comment by Lorenzbi** [comment]
```
viewer
    xwin_id : '58720257'
    cmd_start : 'zathura  -x "/usr/bin/vim --servername  --remote +\%{line} \%
{input}" ''/home/lbischof/latex/build/main.pdf'''
    cmd_forward_search : 'zathura --synctex-forward 1:1:''/home/lbischof/latex
/main.tex'' ''/home/lbischof/latex/build/main.pdf'''
    class : 'Zathura'
    init : function('16')
    view : function('17')
    start : function('18')
    forward_search : function('19')
    latexmk_callback : function('20')
    latexmk_append_argument : function('21')
    hook_view : function('<SNR>52_focus_viewer')
    hook_callback : function('<SNR>52_focus_vim')
    xwin_exists : function('<SNR>52_xwin_exists')
    xwin_get_id : function('<SNR>52_xwin_get_id')
```
I only have these vimtex specific options in my vimrc:
let g:vimtex_latexmk_build_dir="build"
let g:vimtex_view_method="zathura"
And synctex enabled in my latexmkrc

I am using a distro based on ubuntu 14.04, vim without gui 7.4 patch 1-843 without clientserver. xdotool version 3.20140217.1

**Comment by lervag** [comment]
Thanks. And sorry, I meant that in step two, you should do `\ll`, not `\lv`. Could you provide the similar output if you use `\ll` instead of `\lv`?

The reason it does not work is that you do not use the clientserver. The clientserver is used to make latexmk communicate back to vimtex about the viewer.

**Comment by Lorenzbi** [comment]
```
b:vimtex : main
  pid : 15878
  root : '/home/lbischof/latex'
  aux : '/home/lbischof/latex/build/main.aux'
  log : '/home/lbischof/latex/build/main.log'
  out : '/home/lbischof/latex/build/main.pdf'
  tex : '/home/lbischof/latex/main.tex'
  tmp : '/tmp/user/1001/vSKOrby/3'
  cmd_latexmk_compile : 'cd ''/home/lbischof/latex'' && max_print_line=2000 la
texmk -pdf -e ''$pdflatex =~ s/ / -file-line-error /'' -outdir=build -pvc -e '
'$new_viewer_always = "0"'' -e ''$pdf_previewer = "zathura  -x \"/usr/bin/vim
--servername  --remote +\%{line} \%{input}\" \%S"'' ''main.tex'' >/tmp/user/10
01/vSKOrby/3 2>&1'
  base : 'main.tex'
  viewer
    xwin_id : 0
    class : 'Zathura'
    init : function('16')
    view : function('17')
    start : function('18')
    forward_search : function('19')
    latexmk_callback : function('20')
    latexmk_append_argument : function('21')
    hook_view : function('<SNR>52_focus_viewer')
    hook_callback : function('<SNR>52_focus_vim')
    xwin_exists : function('<SNR>52_xwin_exists')
    xwin_get_id : function('<SNR>52_xwin_get_id')
```
I'm not using clientserver because I would need to compile vim myself (as far as I know). Is there anyway to disable the viewer on compile since it works if I just use \lv? Would my problem be solved if I would use clientserver?

**Comment by lervag** [comment]
Yes, the problem should be solved if you enable the clientserver. It really is not very difficult to compile Vim yourself. Also, I think it you install vim with gui, then you will also get the clientserver (you don't need to use the gui).

However, I think I could also add an option that makes latexmk not open a viewer. If you insist, then I will consider it.

**Comment by Lorenzbi** [comment]
I just installed vim-gnome and it worked. Then this issue could be closed. Maybe this could be included in the documentation? Is your goal to also support vimtex without clientserver? If yes, then it would be great to also have an option to customize the compile.

**Comment by lervag** [comment]
I've already included in the documentation that the clientserver is needed for callback functionality. In general, I don't want to support every corner case, and I would be glad if I did not have to add more options. So if you are happy with using vim-gnome, then I think we'll leave it at that for now.

If anyone insists that I should improve support for Vim with no clientserver, then I will consider it. Until then, I will instead insist that people use the clientserver. :)
